tscached/handler_general.py

```diff
@@ -56,37 +50,14 @@ def handle_query():
         # readahead shadow load support
         process_for_readahead(config, redis_client, kquery.get_key(), request.referrer, request.headers)
 
-        if kq_result:
-            try:
-                start_cache = datetime.datetime.fromtimestamp(float(kq_result['earliest_data']))
-                end_cache = datetime.datetime.fromtimestamp(float(kq_result['last_add_data']))
-            except:
-                # some sort of cache malformation or error
-                start_cache = None
-                end_cache = None
-            staleness_threshold = 10  # TODO static lookup in config
-            range_needed = get_range_needed(start_request, end_request, start_cache,
-                                            end_cache, staleness_threshold)
-
-            if not range_needed:
-                # hot / hit
-                kq_resp = cache_calls.hot(redis_client, kquery, kairos_time_range)
+        try:
+            if kq_result:
+                kq_resp = cache_calls.process_cache_hit(config, redis_client, kquery, kairos_time_range)
             else:
-                merge_method = range_needed[2]
-                if merge_method == FETCH_ALL:
-                    logging.info('Odd COLD scenario: data exists.')
-                    # cold / miss
-                    kq_resp = cache_calls.cold(config, redis_client, kquery, kairos_time_range)
-                elif merge_method in [FETCH_BEFORE, FETCH_AFTER]:
-                    # warm / stale
-                    kq_resp = cache_calls.warm(config, redis_client, kquery, kairos_time_range,
-                                               range_needed)
-                else:
-                    logging.error("Received an unsupported range_needed value: %s" % range_needed[2])
-                    kq_resp = {}
-        else:
-            # complete redis miss: cold
-            kq_resp = cache_calls.cold(config, redis_client, kquery, kairos_time_range)
+                kq_resp = cache_calls.cold(config, redis_client, kquery, kairos_time_range)
+        except BackendQueryFailure as e:
+            logging.error('BackendQueryFailure: %s' % e.message)
+            return json.dumps({'error': e.message}), 500
```

Comment:
Should the response code from the BackendQueryFailure object be used here? :) (with the presumption that "requests.exceptions.RequestException" corresponds to any error and not just a 500)